tracer/src/Datadog.Tracer.Native/debugger_method_rewriter.cpp

```diff
@@ -1418,12 +1457,14 @@ HRESULT DebuggerMethodRewriter::Rewrite(RejitHandlerModule* moduleHandler,
 
         if (hr != E_NOTIMPL && FAILED(hr))
         {
+            MarkAllProbesAsError(methodProbes, lineProbes, invalid_probe_failed_to_instrument_line_probe);
```

Comment:
The comment is not specific to this line but in general.
Since we reuse the same `hr` variable, don't we mark "good" probes with error? e.g. if line probe has succeeded and than a method probe failed, we send all probes to `MarkAllProbesAsError`, is it ok?

Reply:
I understand but still won't be a case where a method probe completely fails for some reason but we can still place a line probe? Maybe I'm missing something here, we can discuss it offline

Reply:
It we fail to place line / method probe with `E_FAIL` then we avoid instrumenting the method altogether. Note that `E_FAIL` is too sensitive to try to keep instrumenting. `E_NOTIMPL` on the other hand is where we tolerate the failure and keep instrumenting. It might be that we may also tolerate various `E_FAIL` but we need to look at each and every step where we may encounter `E_FAIL` in Method Probe (async & non-async) instrumentation.
We may consider doing it after the Fault-Tolerant instrumentation will be implemented.
WDYT?